### PR TITLE
Secrets Panel button: Anonymous Names (and refactors a bit of admin code)

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Secrets button next to AI Triumvirate that makes all the names for people anonymous.

Normal crewmembers become "Employee Q5460Z"
Silicons become "Employee Assistant/Core Assuming Delta"

## Why It's Good For The Game

This was talked about in the admin channels as a really cool button idea, one where everyone becomes generic and anonymous. While it has no place as a permanent option or something that even should happen (degradation of roleplay) it serves as a really cool event tool.

## Changelog
:cl:
admin: new secrets button, anonymous names (roundstart required)
refactor: things are a bit cleaner in the AI Triumvirate code
/:cl:

### CODE REVIEW

**Comment by Iamgoofball** [comment]
i'd vote for this

**Comment by EOBGames** [comment]
I'd buy that for a dollar

**Comment by Rohesie** [comment]
Pool of human-readable names > random numbers and letters.
It's interesting to have round-start anonymity (and you also need to scramble the visual), but this just makes it a pain to report or follow up on crimes.

**Comment by tralezab** [comment]
Here, i'll add an option for random names + defaults for AI if you want to use that instead.

File: code/controllers/subsystem/ticker.dm [hunk 1]
```diff
@@ -27,7 +27,8 @@ SUBSYSTEM_DEF(ticker)
 	var/admin_delay_notice = ""				//a message to display to anyone who tries to restart the world after a delay
 	var/ready_for_reboot = FALSE			//all roundend preparation done with, all that's left is reboot
 
-	var/triai = 0							//Global holder for Triumvirate
+	var/anonymousnames = ANON_DISABLED		//Do people spawn with anonymous names? If not ANON_DISABLED then people spawn with a themed anon name (see anonymousnames.dm)
+	var/triai = FALSE						//Global holder for Triumvirate
```
Comment (bobbahbrown) [inline code/controllers/subsystem/ticker.dm:31]: dmdoc it

File: code/modules/admin/verbs/anonymousnames.dm [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+/**
+  * anonymous_name: generates a corporate random name. used in admin event tool anonymous names
+  *
+  * first letter is always a letter
+  * Example name = "Employee Q5460Z"
+  * Arguments:
+  * * M - mob for preferences and gender
+  */
+
```
Comment (bobbahbrown) [inline code/modules/admin/verbs/anonymousnames.dm:35]: remove newlines between commennt and thing commented or else dmdoc doesnt work!!!!

File: code/modules/admin/verbs/anonymousnames.dm [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		if(ANON_RANDOMNAMES)
+			return M.client.prefs.pref_species.random_name(M.gender,1)
+		if(ANON_EMPLOYEENAMES)
+			var/name = "Employee "
+
+			for(var/i in 1 to 6)
+				if(prob(30) || i == 1)
+					name += ascii2text(rand(65, 90)) //A - Z
+				else
+					name += ascii2text(rand(48, 57)) //0 - 9
```
Comment (bobbahbrown) [inline code/modules/admin/verbs/anonymousnames.dm:47]: you could use a trinary operator here to significantly minify this ( <e> ? <f> : <g> ) but you may lose the commentary you are trying to attain here so up to you

File: code/modules/admin/verbs/anonymousnames.dm [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+		to_chat(usr, "This option is currently only usable during pregame.")
+		return
+
+	if(SSticker.anonymousnames)
+		SSticker.anonymousnames = ANON_DISABLED
+		to_chat(usr, "Disabled anonymous names.")
+		message_admins("<span class='adminnotice'>[key_name_admin(usr)] has disabled anonymous names.</span>")
+		return
+	var/list/names = list("Cancel", ANON_RANDOMNAMES, ANON_EMPLOYEENAMES)
+	var/result = input(usr, "Choose an anonymous theme","going dark") as null|anything in names
```
Comment (bobbahbrown) [inline code/modules/admin/verbs/anonymousnames.dm:16]: note this is not a stripped input

just sayin

File: code/modules/admin/verbs/tripAI.dm [hunk 5]
```diff
@@ -1,20 +1,20 @@
...
+		to_chat(usr, "Unable to locate the AI job", confidential = TRUE)
+		return
+	if(SSticker.triai)
+		SSticker.triai = FALSE
+		to_chat(usr, "Only one AI will be spawned at round start.")
+		message_admins("<span class='adminnotice'>[key_name_admin(usr)] has toggled off triple AIs at round start.</span>")
+	else
+		SSticker.triai = TRUE
+		to_chat(usr, "There will be an AI Triumvirate at round start.")
+		message_admins("<span class='adminnotice'>[key_name_admin(usr)] has toggled on triple AIs at round start.</span>")
```
Comment (bobbahbrown) [inline code/modules/admin/verbs/tripAI.dm:20]: you could make this 3 lines if you just change the wording of those two messages and use trinary operators (a ? b : c) to choose the will/will not etc words

File: code/modules/admin/verbs/anonymousnames.dm (hunk 4 above)
Comment (tralezab) [reply code/modules/admin/verbs/anonymousnames.dm:16]: ??? it's from the list

File: code/modules/admin/verbs/tripAI.dm (hunk 5 above)
Comment (tralezab) [reply code/modules/admin/verbs/tripAI.dm:20]: very true

File: code/modules/admin/verbs/anonymousnames.dm (hunk 4 above)
Comment (bobbahbrown) [reply code/modules/admin/verbs/anonymousnames.dm:16]: sorry im r*******

thanks for helping

File: code/modules/admin/verbs/anonymousnames.dm (hunk 3 above)
Comment (tralezab) [reply code/modules/admin/verbs/anonymousnames.dm:47]: i really like the readability lets see what a maintainer says

**Comment by EgorDinamit** [comment]
"Citizen Z893 violated code 4-9, Terminate.", "Officer Q283, termination of citizen Z893 confirmed, thank for your report."

**Pull request merged**